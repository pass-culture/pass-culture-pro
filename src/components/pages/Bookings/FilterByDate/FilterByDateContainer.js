import { compose } from 'redux'
import { connect } from 'react-redux'
import { withFrenchQueryRouter } from '../../../hocs'

import FilterByDate from './FilterByDate'
import { selectStocksByOfferId } from '../../../../selectors/data/stocksSelectors'
import { selectOfferById } from '../../../../selectors/data/offersSelectors'
import { selectVenueById } from '../../../../selectors/data/venuesSelectors'

export const mapDispatchToProps = dispatch => ({
  updateBookingsFrom: date => {
    dispatch({
      payload: date,
      type: 'BOOKING_SUMMARY_UPDATE_BOOKINGS_FROM',
    })
  },
  updateBookingsTo: date => {
    dispatch({
      payload: date,
      type: 'BOOKING_SUMMARY_UPDATE_BOOKINGS_TO',
    })
  },
})

const getTimezoneFromDepartementCode = departementCode => {
  switch (departementCode) {
    case '97':
    case '973':
      return 'America/Cayenne'
    default:
      return 'Europe/Paris'
  }
}

export const mapStateToProps = state => {
  const { bookingSummary = {} } = state
  const { offerId, venueId } = bookingSummary

  let stocks = []
  let showEventDateSection = false
  let showThingDateSection = false
  let timezone = 'Europe/Paris'

  if (offerId && offerId !== 'all') {
    const offer = selectOfferById(state, offerId)

    if (offer.isThing) {
      showThingDateSection = true
    }

    if (offer.isEvent) {
      const venue = selectVenueById(state, venueId)
      if (!venue.isVirtual) {
        timezone = getTimezoneFromDepartementCode(venue.departementCode)
      }
      stocks = selectStocksByOfferId(state, offerId)
      showEventDateSection = true
    }
  }

  return {
    showEventDateSection,
    showThingDateSection,
    stocks,
    timezone,
  }
}

export default compose(
  withFrenchQueryRouter,
  connect(
    mapStateToProps,
    mapDispatchToProps
  )
)(FilterByDate)
