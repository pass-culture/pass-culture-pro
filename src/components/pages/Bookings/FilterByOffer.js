import React, { PureComponent } from 'react'
import classnames from 'classnames'
import PropTypes from 'prop-types'

export class FilterByOffer extends PureComponent {

  onChangeOffer = event => {
    const selectedOffer = document.getElementById("offers")
    const offerId = selectedOffer[selectedOffer.selectedIndex].value
    console.log("OfferId", offerId)
  }

  onChangeDigitalOffer = event => {
    const selectedDigitalOffer = document.getElementById("offers")
    const digitalOfferId = selectedDigitalOffer[selectedDigitalOffer.selectedIndex].value
    console.log("OfferId", digitalOfferId)
  }


  render() {
    const { offers } = this.props
    const isDigitalChecked = true

    if (!isDigitalChecked){
      return (
        <React.Fragment>
          <div id="filter-by-offer"
               className={classnames({'is-invisible': isDigitalChecked,})} >
            <div className="section">
              <h1 className="main-list-title">
                OFFRE
              </h1>
            </div>
            <h2>
              {'Sélectionner les réservations pour l\'offre :'}
            </h2>
            <select
              onChange={this.onChangeOffer}
              className="pc-selectbox pl24 py5 fs19"
              defaultValue="all"
              id="offers"
            >
              {offers.map(({ name, id }) => (
                <option key={id} value={id}>
                  {name}
                </option>
              ))}
            </select>
          </div>
        </React.Fragment>
      )
    }

    return (
      <React.Fragment>
        <div id="filter-by-digital-offer"
             className={classnames({'is-invisible': !isDigitalChecked,})}>
          <div className="section">
            <h1 className="main-list-title">
              OFFRE NUMERIQUE
            </h1>
          </div>
          <h2>
            {'Sélectionner les réservations pour l\'offre numérique :'}
          </h2>
          <select
            onChange={this.onChangeDigitalOffer}
            className="pc-selectbox pl24 py5 fs19"
            defaultValue="all"
            id="offers"
          >
            {offers.map(({ name, id }) => (
              <option key={id} value={id}>
                {name}
              </option>
            ))}
          </select>
        </div>
      </React.Fragment>
    )
  }
}

FilterByOffer.propTypes = {
  offers: PropTypes.array.isRequired,
}

