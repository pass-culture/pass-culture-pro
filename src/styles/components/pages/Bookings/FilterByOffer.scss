/* ------------------------------------------------

 Selection des offres

------------------------------------------------ */

#filter-by-offer
div {
  font-size: 16px;
  margin-bottom: rem(6px);
}

#filter-by-offer .pc-selectbox {
  @extend .no-outline;

  border-radius: 18px;
  width: 488px;
  height: 36px;
  background-repeat: no-repeat;
  background-position: 455px 14px;
  background-image: url(/icons/dropdown-disclosure-down-s-w.svg);
  background-color: transparent;
  box-shadow: 0 2px 6px 0 rgba(0,0,0,0.25);
  font-size: 16px;
  margin-bottom: rem(18px);
  -webkit-appearance: none;
  -moz-appearance: none;
}

/* ------------------------------------------------

 Selection des offres numeriques

------------------------------------------------ */

#filter-by-digital-offer
div {
  font-size: 16px;
  margin-bottom: rem(6px);
}

#filter-by-digital-offer .pc-selectbox {
  @extend .no-outline;

  border-radius: 18px;
  width: 488px;
  height: 36px;
  background-repeat: no-repeat;
  background-position: 455px 14px;
  background-image: url(/icons/dropdown-disclosure-down-s-w.svg);
  background-color: transparent;
  box-shadow: 0 2px 6px 0 rgba(0,0,0,0.25);
  font-size: 16px;
  margin-bottom: rem(18px);
  -webkit-appearance: none;
  -moz-appearance: none;
}
