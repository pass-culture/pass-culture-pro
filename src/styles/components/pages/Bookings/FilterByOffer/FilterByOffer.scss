#filter-by-offer {
  .pc-selectbox {

    font-size: 16px;
    border-radius: 18px;
    width: 488px;
    height: 36px;
    background-repeat: no-repeat;
    background-position: 455px 14px;
    background-image: url(/icons/dropdown-disclosure-down-s-w.svg);
    background-color: transparent;
    box-shadow: 0 2px 6px 0 rgba(0, 0, 0, 0.25);
    margin: rem(6px) 0 rem(18px) 0;
    -webkit-appearance: none;
    -moz-appearance: none;
  }

  label {
    font-size: 16px;
  }
}
