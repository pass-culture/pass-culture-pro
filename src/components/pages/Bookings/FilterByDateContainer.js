import {compose} from 'redux'
import {connect} from 'react-redux'
import {withFrenchQueryRouter} from 'components/hocs'

import {FilterByDate} from './FilterByDate'

export const mapDispatchToProps = (dispatch) => ({
  selectBookingsForDate: (date) => {
    dispatch(
      {
        payload: date,
        type: 'BOOKING_SUMMARY_SELECT_DATE',
      }
    )
  }
})

export const mapStateToProps = (state) => {
  return {
    date: state.bookingSummary.selectOffersSince,
  }
}

export default compose(
  withFrenchQueryRouter,
  connect(mapStateToProps, mapDispatchToProps)
)(FilterByDate)
