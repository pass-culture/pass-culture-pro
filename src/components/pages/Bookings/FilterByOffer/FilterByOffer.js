import React, { Fragment, PureComponent } from 'react'
import PropTypes from 'prop-types'

import FilterByDateContainer from '../FilterByDate/FilterByDateContainer'

export class FilterByOffer extends PureComponent {
  componentDidMount() {
    const { loadOffers } = this.props
    loadOffers()
  }

  render() {
    const {
      isFilterByDigitalVenues,
      offerId,
      offersOptions,
      selectBookingsForOffers,
      showDateSection,
    } = this.props

    return (
      <Fragment>
        <div className="section">
          <h2 className="main-list-title">
            {isFilterByDigitalVenues ? 'Offre numérique' : 'Offre'}
          </h2>
        </div>
        <div id="filter-by-offer">
          <label htmlFor="offers">
            {"Télécharger les réservations pour l'offre "}
            {isFilterByDigitalVenues ? 'numérique' : ''}
            {':'}
          </label>
          <select
            className="pc-selectbox pl24 py5 fs19"
            id="offers"
            onBlur={selectBookingsForOffers}
            onChange={selectBookingsForOffers}
            value={offerId}
          >
            <option
              disabled
              label=" "
            />
            {offersOptions.map(({ name, id }) => (
              <option
                key={id}
                value={id}
              >
                {name}
              </option>
            ))}
          </select>
        </div>
        {showDateSection && <FilterByDateContainer />}
      </Fragment>
    )
  }
}

FilterByOffer.defaultProps = {
  offerId: '',
}

FilterByOffer.propTypes = {
  isFilterByDigitalVenues: PropTypes.bool.isRequired,
  loadOffers: PropTypes.func.isRequired,
  offerId: PropTypes.string,
  offersOptions: PropTypes.arrayOf(PropTypes.shape()).isRequired,
  selectBookingsForOffers: PropTypes.func.isRequired,
  showDateSection: PropTypes.bool.isRequired,
}
