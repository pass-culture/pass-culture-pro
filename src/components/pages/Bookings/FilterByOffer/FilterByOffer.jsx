import React, { Fragment, PureComponent } from 'react'
import PropTypes from 'prop-types'

import FilterByDateContainer from '../FilterByDate/FilterByDateContainer'

class FilterByOffer extends PureComponent {
  componentDidMount() {
    const { loadOffers } = this.props
    loadOffers()
  }

  render() {
    const {
      isFilteredByDigitalVenues,
      offerId,
      offersOptions,
      showDateSection,
      updateOfferId,
    } = this.props

    return (
      <Fragment>
        <div className="section">
          <h2 className="main-list-title">
            {isFilteredByDigitalVenues ? 'Offre numérique' : 'Offre'}
          </h2>
        </div>
        <div id="filter-by-offer">
          <label htmlFor="offers">
            {"Télécharger les réservations pour l'offre "}
            {isFilteredByDigitalVenues ? 'numérique ' : ''}
            {':'}
          </label>
          <select
            className="pc-selectbox"
            id="offers"
            onBlur={updateOfferId}
            onChange={updateOfferId}
            value={offerId}
          >
            <option
              disabled
              label=" - Choisissez une offre dans la liste - "
              selected
            />
            {offersOptions.map(({ name, id }) => (
              <option
                key={id}
                value={id}
              >
                {name}
              </option>
            ))}
          </select>
        </div>
        {showDateSection && <FilterByDateContainer />}
      </Fragment>
    )
  }
}

FilterByOffer.defaultProps = {
  offerId: '',
}

FilterByOffer.propTypes = {
  isFilteredByDigitalVenues: PropTypes.bool.isRequired,
  loadOffers: PropTypes.func.isRequired,
  offerId: PropTypes.string,
  offersOptions: PropTypes.arrayOf(PropTypes.shape()).isRequired,
  showDateSection: PropTypes.bool.isRequired,
  updateOfferId: PropTypes.func.isRequired,
}

export default FilterByOffer
