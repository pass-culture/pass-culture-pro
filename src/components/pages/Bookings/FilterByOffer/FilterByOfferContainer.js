import { connect } from 'react-redux'
import { requestData } from 'redux-saga-data'

import FilterByOffer from './FilterByOffer'
import selectDigitalOffers from '../selectors/selectDigitalOffers'
import selectOffersByVenueId from '../selectors/selectOffersByVenueId'
import selectIsUserAdmin from '../selectors/selectIsUserAdmin'

export const mapDispatchToProps = dispatch => ({
  loadOffers: () => {
    dispatch(
      requestData({
        apiPath: `/offers`,
        method: 'GET',
        stateKey: 'offers',
      })
    )
  },
  updateOfferId: event => {
    dispatch({
      payload: event.target.value,
      type: 'BOOKING_SUMMARY_UPDATE_OFFER_ID',
    })
  },
})

export const mapStateToProps = state => {
  const isUserAdmin = selectIsUserAdmin(state)

  const { bookingSummary = {} } = state
  const { isFilteredByDigitalVenues, offerId, venueId } = bookingSummary

  const allOffersOption = {
    id: 'all',
    name: 'Toutes les offres',
  }

  let offersOptions = []

  if (!isUserAdmin) {
    if (isFilteredByDigitalVenues) {
      const digitalOffers = selectDigitalOffers(state)
      offersOptions = [allOffersOption, ...digitalOffers]
    } else {
      if (venueId === 'all') {
        offersOptions = []
      } else {
        const offersFromSpecificVenue = selectOffersByVenueId(state, venueId)
        offersOptions = [allOffersOption, ...offersFromSpecificVenue]
      }
    }
  }

  const showDateSection = offerId !== 'all' && !!offerId

  return {
    isFilteredByDigitalVenues,
    offerId,
    offersOptions,
    showDateSection,
    venueId,
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(FilterByOffer)
