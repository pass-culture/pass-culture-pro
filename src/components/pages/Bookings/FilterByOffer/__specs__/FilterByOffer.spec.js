import React from 'react'
import { shallow } from 'enzyme/build'

import FilterByDateContainer from '../../FilterByDate/FilterByDateContainer'
import FilterByOffer from '../FilterByOffer'

describe('src | components | pages | FilterByOffer | FilterByOffer', () => {
  let props

  beforeEach(() => {
    props = {
      isFilteredByDigitalVenues: false,
      loadOffers: jest.fn(),
      offers: [],
      offersOptions: [
        {
          id: 'all',
          name: 'Toutes les offres',
        },
        {
          id: 'CY',
          name: 'Spectacle de danse',
        },
      ],
      updateOfferId: jest.fn(),
      showDateSection: false,
    }
  })

  it('should match snapshot', () => {
    // when
    const wrapper = shallow(<FilterByOffer {...props} />)

    // then
    expect(wrapper).toMatchSnapshot()
  })

  describe('render', () => {
    describe('the date section', () => {
      it('should be hidden when `all offer is selected', () => {
        // when
        const wrapper = shallow(<FilterByOffer {...props} />)

        // then
        const filterByDateContainer = wrapper.find(FilterByDateContainer)
        expect(filterByDateContainer).toHaveLength(0)
      })

      it('should display offer section when a specific offer is selected', () => {
        // given
        props.showDateSection = true

        // when
        const wrapper = shallow(<FilterByOffer {...props} />)

        // then
        const filterByDateContainer = wrapper.find(FilterByDateContainer)
        expect(filterByDateContainer).toHaveLength(1)
      })
    })

    describe('labels', () => {
      it('should display `Offre` when isFilteredByDigitalVenues is false', () => {
        // given
        props.isFilteredByDigitalVenues = false

        // when
        const wrapper = shallow(<FilterByOffer {...props} />)

        // then
        const mainListTitle = wrapper.find('.main-list-title').text()
        expect(mainListTitle).toBe('Offre')
      })

      it('should display `Offre numérique` when isFilteredByDigitalVenues is true', () => {
        // given
        props.isFilteredByDigitalVenues = true

        // when
        const wrapper = shallow(<FilterByOffer {...props} />)

        // then
        const mainListTitle = wrapper.find('.main-list-title').text()
        expect(mainListTitle).toBe('Offre numérique')
      })

      it("should display `Télécharger les réservations pour l'offre :` when isFilteredByDigitalVenues is false", () => {
        // given
        props.isFilteredByDigitalVenues = false

        // when
        const wrapper = shallow(<FilterByOffer {...props} />)

        // then
        const label = wrapper.find('label')
        expect(label.at(0).text()).toBe("Télécharger les réservations pour l'offre :")
      })

      it("should display `Télécharger les réservations pour l'offre numérique:` when isFilteredByDigitalVenues is true", () => {
        // given
        props.isFilteredByDigitalVenues = true

        // when
        const wrapper = shallow(<FilterByOffer {...props} />)

        // then
        const label = wrapper.find('label')
        expect(label.at(0).text()).toBe("Télécharger les réservations pour l'offre numérique :")
      })
    })

    describe('selector', () => {
      it('should render a select input for offer with 3 options', () => {
        // when
        const wrapper = shallow(<FilterByOffer {...props} />)

        // then
        const selectOffer = wrapper.find('#offers')
        expect(selectOffer.prop('className')).toBe('pc-selectbox')
        expect(selectOffer.prop('id')).toBe('offers')
        expect(selectOffer.prop('onBlur')).toStrictEqual(expect.any(Function))
        expect(selectOffer.prop('onChange')).toStrictEqual(expect.any(Function))
        const options = selectOffer.find('option')
        expect(options).toHaveLength(3)
        expect(options.at(0).prop('disabled')).toBe(true)
        expect(options.at(0).prop('label')).toBe(' - Choisissez une offre dans la liste - ')
        expect(options.at(0).prop('selected')).toBe(true)
        expect(options.at(1).key()).toBe('all')
        expect(options.at(1).prop('value')).toBe('all')
        expect(options.at(2).key()).toBe('CY')
        expect(options.at(2).prop('value')).toBe('CY')
      })
    })
    
  })
})
