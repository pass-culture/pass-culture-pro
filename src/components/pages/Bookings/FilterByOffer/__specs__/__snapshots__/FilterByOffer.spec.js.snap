// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src | components | pages | FilterByOffer | FilterByOffer should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FilterByOffer
    isFilteredByDigitalVenues={false}
    loadOffers={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    offerId=""
    offers={Array []}
    offersOptions={
      Array [
        Object {
          "id": "all",
          "name": "Toutes les offres",
        },
      ]
    }
    showDateSection={false}
    updateOfferId={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div
          className="section"
        >
          <h2
            className="main-list-title"
          >
            Offre
          </h2>
        </div>,
        <div
          id="filter-by-offer"
        >
          <label
            htmlFor="offers"
          >
            Télécharger les réservations pour l'offre 
            
            :
          </label>
          <select
            className="pc-selectbox"
            id="offers"
            onBlur={[MockFunction]}
            onChange={[MockFunction]}
            value=""
          >
            <option
              disabled={true}
              label=" - Choisissez une offre - "
              selected={true}
            />
            <option
              value="all"
            >
              Toutes les offres
            </option>
          </select>
        </div>,
        false,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h2
            className="main-list-title"
          >
            Offre
          </h2>,
          "className": "section",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Offre",
            "className": "main-list-title",
          },
          "ref": null,
          "rendered": "Offre",
          "type": "h2",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              htmlFor="offers"
            >
              Télécharger les réservations pour l'offre 
              
              :
            </label>,
            <select
              className="pc-selectbox"
              id="offers"
              onBlur={[MockFunction]}
              onChange={[MockFunction]}
              value=""
            >
              <option
                disabled={true}
                label=" - Choisissez une offre - "
                selected={true}
              />
              <option
                value="all"
              >
                Toutes les offres
              </option>
            </select>,
          ],
          "id": "filter-by-offer",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Télécharger les réservations pour l'offre ",
                "",
                ":",
              ],
              "htmlFor": "offers",
            },
            "ref": null,
            "rendered": Array [
              "Télécharger les réservations pour l'offre ",
              "",
              ":",
            ],
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <option
                  disabled={true}
                  label=" - Choisissez une offre - "
                  selected={true}
                />,
                Array [
                  <option
                    value="all"
                  >
                    Toutes les offres
                  </option>,
                ],
              ],
              "className": "pc-selectbox",
              "id": "offers",
              "onBlur": [MockFunction],
              "onChange": [MockFunction],
              "value": "",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "disabled": true,
                  "label": " - Choisissez une offre - ",
                  "selected": true,
                },
                "ref": null,
                "rendered": null,
                "type": "option",
              },
              Object {
                "instance": null,
                "key": "all",
                "nodeType": "host",
                "props": Object {
                  "children": "Toutes les offres",
                  "value": "all",
                },
                "ref": null,
                "rendered": "Toutes les offres",
                "type": "option",
              },
            ],
            "type": "select",
          },
        ],
        "type": "div",
      },
      false,
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            className="section"
          >
            <h2
              className="main-list-title"
            >
              Offre
            </h2>
          </div>,
          <div
            id="filter-by-offer"
          >
            <label
              htmlFor="offers"
            >
              Télécharger les réservations pour l'offre 
              
              :
            </label>
            <select
              className="pc-selectbox"
              id="offers"
              onBlur={[MockFunction]}
              onChange={[MockFunction]}
              value=""
            >
              <option
                disabled={true}
                label=" - Choisissez une offre - "
                selected={true}
              />
              <option
                value="all"
              >
                Toutes les offres
              </option>
            </select>
          </div>,
          false,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h2
              className="main-list-title"
            >
              Offre
            </h2>,
            "className": "section",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Offre",
              "className": "main-list-title",
            },
            "ref": null,
            "rendered": "Offre",
            "type": "h2",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                htmlFor="offers"
              >
                Télécharger les réservations pour l'offre 
                
                :
              </label>,
              <select
                className="pc-selectbox"
                id="offers"
                onBlur={[MockFunction]}
                onChange={[MockFunction]}
                value=""
              >
                <option
                  disabled={true}
                  label=" - Choisissez une offre - "
                  selected={true}
                />
                <option
                  value="all"
                >
                  Toutes les offres
                </option>
              </select>,
            ],
            "id": "filter-by-offer",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Télécharger les réservations pour l'offre ",
                  "",
                  ":",
                ],
                "htmlFor": "offers",
              },
              "ref": null,
              "rendered": Array [
                "Télécharger les réservations pour l'offre ",
                "",
                ":",
              ],
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <option
                    disabled={true}
                    label=" - Choisissez une offre - "
                    selected={true}
                  />,
                  Array [
                    <option
                      value="all"
                    >
                      Toutes les offres
                    </option>,
                  ],
                ],
                "className": "pc-selectbox",
                "id": "offers",
                "onBlur": [MockFunction],
                "onChange": [MockFunction],
                "value": "",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "disabled": true,
                    "label": " - Choisissez une offre - ",
                    "selected": true,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": "all",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Toutes les offres",
                    "value": "all",
                  },
                  "ref": null,
                  "rendered": "Toutes les offres",
                  "type": "option",
                },
              ],
              "type": "select",
            },
          ],
          "type": "div",
        },
        false,
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
